var/spack/repos/builtin/packages/libbeagle/package.py

```diff
@@ -29,16 +29,24 @@ class Libbeagle(AutotoolsPackage, CudaPackage):
     def patch(self):
         # update cuda architecture if necessary
         if '+cuda' in self.spec:
-            arch = self.spec.variants['cuda_arch'].value
+            cuda_arch = self.spec.variants['cuda_arch'].value
             archflag = ''
 
-            if arch[0] != 'none':
-                archflag = '-arch=%s' % arch[0]
+            if cuda_arch != 'none':
+                if len(cuda_arch) > 1:
```

Comment:
Is `cuda_arch` a list or a string? Shouldn't it be `cuda_arch[0] != 'none'`?

Reply:
My understanding is that if `cuda_arch` is set to values then it is a list, but if `cuda_arch` is unset it has a value of 'none', which I _think_ is a string. The cuda build system uses `spack.variant.any_combination_of(*cuda_arch_values)` for the `cpu_arch` variant.

From lib/spack/spack/variant.py:
```
def any_combination_of(*values):                                                                                        
    """Multi-valued variant that allows any combination of the specified                                                
    values, and also allows the user to specify 'none' (as a string) to choose                                          
    none of them.
```

Reply:
I looked through the existing packages and things are really inconsistent. We should probably make this more clear in the documentation and update all packages to match, but someone else can do that.